salt/kubernetes-minion/kubelet.jinja

```diff
@@ -12,8 +12,8 @@ KUBELET_ADDRESS="--address=0.0.0.0"
 # The port for the info server to serve on
 KUBELET_PORT="--port=10250"
 
-# You may leave this blank to use the actual hostname
-KUBELET_HOSTNAME=""
+# Use machine ID as hostname to ensure uniqueness
+KUBELET_HOSTNAME="--hostname-override={{ grains['id'] }}"
```

Comment:
Based on the updates to https://trello.com/c/yCbzToby/56-8-regenerate-certificates-when-ip-changes - we should make this just `api.{{ pillar['internal_infra_domain'] }}` - #44 has been updated to support this.